lib/pubid/iso/amendment.rb

```diff
@@ -3,32 +3,9 @@ class Amendment < Supplement
     # @param stage_format_long [Boolean] long or short format for stage rendering
     # @param with_date [Boolean] include date
     def render_pubid(stage_format_long = true, with_date = true)
-      stage = render_pubid_stage
       pubid_number = render_pubid_number(with_date: with_date)
-      case stage.to_s
-      when "DIS"
-        if stage_format_long
-          "/DAmd #{pubid_number}"
-        else
-          "/DAM #{pubid_number}"
-        end
-      when "FDIS"
-        if stage_format_long
-          "/FDAmd #{pubid_number}"
-        else
-          "/FDAM #{pubid_number}"
-        end
-      when "CD"
-        if stage_format_long
-          "/CD Amd #{pubid_number}"
-        else
-          "/CDAM #{pubid_number}"
-        end
-      when ""
-        "/Amd #{pubid_number}"
-      else
-        "/#{stage} Amd #{pubid_number}"
-      end
+
+      "/#{@typed_stage.to_s(stage_format_long)} #{pubid_number}"
```

Comment:
Will this end up with a slash followed by an empty space? It seems that some defensive programming is necessary here.

Reply:
> Will this end up with a slash followed by an empty space? It seems that some defensive programming is necessary here.

@typed_stage.to_s will always return something. Without stage or typed stage it will return "Amd" for amendment.
Actually, we don't need separate classes for Amendment and Corrigendum, it's legacy now.

Reply:
What if this is an "IS"? There is no Typed Stage abbreviation and there should not be any slash, see #151